### PR TITLE
Add AccountTerminatedException for terminated channels

### PR DESCRIPTION
**Pull request by TobiGr**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).
- [x] I agree to create a pull request for [NewPipe](https://github.com/TeamNewPipe/NewPipe) as soon as possible to make it compatible with the changed API.

Adds a new `ContentNotAvailableException`: `AccountTerminatedException` to distinguish between terminated channels. The field `reason` provides more info on why the account has been terminated. This fixes a NPE when getting the feed for a terminated channel in YouTubeFeedExtractor.
- [ ] ~~add `CLOSED_BY_OWNER` reason (I need a dedicated channel for this~~
- [ ] ~~implement in other services~~
- [x] add mocks for tests

### CODE REVIEW

**Comment by TobiGr** [comment]
I could not find more info on deleted and terminated channels in the docs of the other services.

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeChannelExtractorTest.java [hunk 1]
```diff
@@ -48,6 +61,91 @@ public void nonExistentFetch() throws Exception {
...
+
+        @Test(expected = AccountTerminatedException.class)
+        public void accountTerminatedCopyrightFetch() throws Exception {
+            final ChannelExtractor extractor =
+                    YouTube.getChannelExtractor("https://www.youtube.com/channel/UCpExuV8qJMfCaSQNL1YG6bQ");
+            try {
+                extractor.fetchPage();
+            } catch (AccountTerminatedException e) {
+                assertEquals(e.getMessage(),
+                        "This account has been terminated because we received multiple third-party claims of copyright infringement regarding material that the user posted.");
```
Comment (XiangRongLin) [inline extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeChannelExtractorTest.java:143]: `assertEquals` has expected first, actual second 
Same for the other asserts.

see https://junit.org/junit4/javadoc/latest/org/junit/Assert.html#assertEquals(java.lang.Object[],%20java.lang.Object[])

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeChannelExtractorTest.java [hunk 2]
```diff
@@ -48,6 +61,91 @@ public void nonExistentFetch() throws Exception {
...
+
+        @Test(expected = AccountTerminatedException.class)
+        public void accountTerminatedTOSFetch() throws Exception {
+            final ChannelExtractor extractor =
+                    YouTube.getChannelExtractor("https://www.youtube.com/channel/UCTGjY2I-ZUGnwVoWAGRd7XQ");
+            try {
+                extractor.fetchPage();
+            } catch (AccountTerminatedException e) {
+                assertEquals(e.getMessage(),
+                        "This account has been terminated for a violation of YouTube's Terms of Service.");
```
Comment (XiangRongLin) [inline extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeChannelExtractorTest.java:73]: I would remove the assertions for the error message. They are bound to fail very soon, if they change the wording. Only usecase i can think of for them, is directly displaying them to the end-user in which case, the exact wording does not change the correctness of the implementation.

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java [hunk 3]
```diff
@@ -692,6 +689,23 @@ public static void defaultAlertsCheck(final JsonObject initialData) throws Parsi
             final String alertText = getTextFromObject(alertRenderer.getObject("text"));
             final String alertType = alertRenderer.getString("type", EMPTY_STRING);
             if (alertType.equalsIgnoreCase("ERROR")) {
+                if (alertText != null && alertText.contains("This account has been terminated")) {
+                    if (alertText.contains("violation") || alertText.contains("violating")
+                            || alertText.contains("infringement")) {
```
Comment (XiangRongLin) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java:694]: what happens if the localization is changes to non-english. I would assume the error messages are localized

Comment (TiA4f8R) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java:694]: @XiangRongLin Yes, they are, it's the same thing for the error messages from YouTube videos.
I think in the future we need to implement something which stores strings of the error messages for each localization and compare with the value provided by YouTube in the language used for the extraction of YouTube datas.

File: extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeChannelExtractorTest.java (hunk 1 above)
Comment (TobiGr) [reply extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeChannelExtractorTest.java:143]: oh yes

**Review by Stypox — APPROVED** [review]
LGTM

**Comment by Stypox** [comment]
Oh wait, this has to be addressed: `I would remove the assertions for the error message. They are bound to fail very soon, if they change the wording. Only usecase i can think of for them, is directly displaying them to the end-user in which case, the exact wording does not change the correctness of the implementation.`

**Review by Stypox — APPROVED** [review]

**Pull request merged**